### PR TITLE
hera update for 1.7.0

### PR DESCRIPTION
**Pull request by AlexanderRichert-NOAA**
### Summary

This PR updates the Hera config for 1.7.0.

### Testing

Tested building on Hera, incl. building one or two UFSWM cases with Intel and GCC.

### Applications affected

all

### Systems affected

hera

### Dependencies

https://github.com/JCSDA/spack/pull/419

### Issue(s) addressed

https://github.com/JCSDA/spack-stack/issues/1042

### Checklist
- [x] This PR addresses one issue/problem/enhancement, or has a very good reason for not doing so.
- [x] These changes have been tested on the affected systems and applications.
- [x] All dependency PRs/issues have been resolved and this PR can be merged.

### CODE REVIEW

**Review by srherbener — APPROVED** [review]

**Review by RatkoVasic-NOAA — APPROVED** [review]

File: configs/sites/hera/packages.yaml [hunk 1]
```diff
@@ -22,11 +22,14 @@ packages:
       prefix: /apps/oneapi
   openmpi:
     externals:
-    - spec: openmpi@4.1.5~cuda~cxx~cxx_exceptions~java+lustre~memchecker+pmi~static~wrapper-rpath schedulers=slurm
-      prefix: /scratch1/NCEPDEV/jcsda/jedipara/spack-stack/openmpi-4.1.5
+    - spec: openmpi@4.1.6~atomics~cuda~cxx~cxx_exceptions~gpfs~internal-hwloc~internal-libevent~internal-pmix~java+legacylaunchers~lustre~memchecker~openshmem~orterunprefix+pmi+romio+rsh~singularity+static+vt+wrapper-rpath fabrics=ucx schedulers=slurm
       modules:
       - gnu/9.2.0
-      - openmpi/4.1.5
+      - openmpi/4.1.6_gnu9.2.0
```
Comment (natalie-perlin) [inline configs/sites/hera/packages.yaml:28]: It is still with gnu/9.2.0?.. Could we push for using 12.2.0?.

Comment (AlexanderRichert-NOAA) [reply configs/sites/hera/packages.yaml:28]: I assume you mean 13.2.0? I'd prefer not to make a last-minute change unless it's critically needed for an application. Could it wait until the 1.8.0 release?

Comment (natalie-perlin) [reply configs/sites/hera/packages.yaml:28]: There is gnu/12.2.0 version installed by Dom, under spack-stack. As I remember, there were some issues with gnu/13.2.0, which is installed system-wide on Hera. 

Way to load module gnu/12.2.0:

```
module use /scratch1/NCEPDEV/nems/role.epic/modulefiles
module load gnu/12.2.0
```

Modulefile `/scratch1/NCEPDEV/nems/role.epic/modulefiles/gnu/12.2.0.lua` was prepared to point to Dom's installation under 
```
/scratch1/NCEPDEV/nems/role.epic/spack-stack/gcc-12.2.0
```

Comment (AlexanderRichert-NOAA) [reply configs/sites/hera/packages.yaml:28]: Ah okay I see, thanks. Would you be willing to update the Hera config to add the second gcc version?

Comment (natalie-perlin) [reply configs/sites/hera/packages.yaml:28]: @AlexanderRichert-NOAA - I'd give it a try!

Comment (climbfuji) [reply configs/sites/hera/packages.yaml:28]: @RatkoVasic-NOAA was also building the gnu@12.2.0 compiler and openmpi@4.1.6 with it - if his testing is successful, we should use that version.

**Pull request merged**